bindings/matrix-sdk-ffi/src/client.rs

```diff
@@ -1002,6 +1002,16 @@ impl Client {
 
         Ok(RoomPreview::from_sdk(sdk_room_preview))
     }
+
+    /// Waits until an at least partially synced room is received, and returns
+    /// it.
+    ///
+    /// **Note: this function will loop endlessly until either it finds the room
+    /// or an externally set timeout happens.**
```

Comment:
i'd do a slightly different API at the FFI level: add an `Option<Duration>` as a parameter, and add a timeout here, so the consumer doesn't have to. Does it make sense? (You get to decide if it's a good idea, based on your usage of the API in the app :))

Reply:
I asked @stefanceriu and we decided to have the client implement the timeout.